dspy/clients/lm.py

```diff
@@ -212,47 +219,82 @@ def copy(self, **kwargs):
         return new_instance
 
 
-@functools.lru_cache(maxsize=None)
-def cached_litellm_completion(request, num_retries: int):
+def request_cache(maxsize: Optional[int] = None):
```

Comment:
@okhat @bahtman @CyrusNuevoDia Thoughts on this approach? See inline comments discussing advantages below

Reply:
Looks cool! 

Could set default `maxsize = float("inf")` here